### ISSUE TITLE
Cluster support

### ISSUE THREAD

**Issue by bakape** [bakape/meguca#251]
* Enable multithreaded scaling through the cluster module
* Use custom sticky session implementation
  * Possibly hash IPs with C++
* Separate server into independent master, 1 upkeep, 1 imager and n worker modules, where n is number of cores. -1.  Upkeep will do miscellaneous timed functions.

**Comment by bakape** [comment]
* Smooth and transparent slave replacement 

Depends on: #251 #162

**Comment by bakape** [comment]
No longer relevant for v2, which will be written in Go.
